### PR TITLE
Limit the number of verticle HTTP event loops based on availble memory

### PR DESCRIPTION
**Pull request by stuartwdouglas**

### CODE REVIEW

File: extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java [hunk 1]
```diff
@@ -233,6 +233,19 @@ public void handle(AsyncResult<Void> event) {
                 "Listening on: http://%s:%s", httpServerOptions.getHost(), httpServerOptions.getPort()));
     }
 
+    private static int calculateDefaultIOThreads() {
+        //we only allow one event loop per 4mb of ram at the most
+        //its hard to say what this number should be, but it is also obvious
+        //that for constrained environments we don't want a lot of event loops
+        //lets start with 4mb and adjust as needed
+        int recommended = Runtime.getRuntime().availableProcessors() * 2;
```
Comment (dmlloyd) [inline extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java]: This should probably use `org.wildfly.common.cpu.ProcessorInfo#availableProcessors` at least for now.

Comment (gsmet) [reply extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java]: @stuartwdouglas do you plan to address this comment?

File: extensions/vertx-core/runtime/src/main/java/io/quarkus/vertx/core/runtime/VertxCoreRecorder.java [hunk 2]
```diff
@@ -12,6 +12,8 @@
 import java.util.regex.Pattern;
 import java.util.stream.Collectors;
 
+import org.wildfly.common.cpu.ProcessorInfo;
```
Comment (cescoffier) [inline extensions/vertx-core/runtime/src/main/java/io/quarkus/vertx/core/runtime/VertxCoreRecorder.java]: Vert.x has something very similar. As this one does not add a dependency, we can keep this one. If one day the dependency is removed, we have a fallback.

File: extensions/vertx-core/runtime/src/main/java/io/quarkus/vertx/core/runtime/VertxCoreRecorder.java [hunk 3]
```diff
@@ -149,6 +153,19 @@ private static VertxOptions convertToVertxOptions(VertxConfiguration conf) {
         return options;
     }
 
+    private static int calculateDefaultIOThreads() {
```
Comment (cescoffier) [inline extensions/vertx-core/runtime/src/main/java/io/quarkus/vertx/core/runtime/VertxCoreRecorder.java]: We also need to reduce the number of verticle deployed.

Event loops are created lazily (but it's always good to cap). when deploying a verticle, an event loop is assigned to it, which trigger the creation if the max is not reached. If you have more verticle than event loops, some verticle would share the event loop.

**Review by cescoffier — DISMISSED** [review]
Asking about the number of verticle as it's related to the number of event loop.

**Review by gsmet — APPROVED** [review]
I checked with Stuart and @cescoffier had a look at this PR and said it was fine.

I'm approving and merging it so that we can have it in 0.23.0.

**Pull request merged**